frontend/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx

```diff
@@ -428,19 +428,19 @@ function getDataArray(
   }
 
   const dataArr = []
-  const [rows, cols] =
-    dataProto.data.length > 0
-      ? [dataProto.data.length, dataProto.data[0].length]
-      : [0, 0]
+  const rows = dataProto.data.length
+  const cols = dataProto.data[0].length
 
-  const indexType = dataProto.types.index[0].name
-  const hasSupportedIndex = SUPPORTED_INDEX_TYPES.has(indexType)
+  const indexType = Quiver.getType(dataProto.types.index[0])
+  const hasSupportedIndex = SUPPORTED_INDEX_TYPES.has(
+    indexType as IndexTypeName
+  )
 
   for (let rowIndex = startIndex; rowIndex < rows; rowIndex++) {
     const row: { [field: string]: any } = {}
 
     if (hasSupportedIndex) {
-      row[MagicFields.DATAFRAME_INDEX] = dataProto.index[0][rowIndex]
+      ;[row[MagicFields.DATAFRAME_INDEX]] = dataProto.index[rowIndex]
```

Comment:
Possible typo? Presumably this should just be `row[MagicFields.DATAFRAME_INDEX]` (with the extra ; and [])

Reply:
That's a destructuring assignment

Reply:
That's some wacky syntax! (Maybe just add a comment that this is intentional?)

Reply:
ESLint forces it